_posts/2025-08-16-accessible-error-messages-in-forms.md

```diff
@@ -0,0 +1,158 @@
+
+{: .example }
+For example: users recognize radio buttons as circles and checkboxes as squares. They know from experience they can choose one option from the radio buttons and multiple options from the checkboxes. Also, the [keyboard interaction](https://webaim.org/techniques/keyboard/#testing) for radiobuttons and checkboxes differs from each other. 
+
+Some field types are better avoided due to poor usability, such as multi-selects. Using date pickers? Check the experience for keyboard- or screen reader users.
+
+Sometimes a simple solution works best for everyone — for example, KLM’s method for selecting a date with a text input for the day, a dropdown for the month, and another text input for the year:
+![Date selection with a text field for day, a select for month, and a text field for year](https://raw.githubusercontent.com/nl-design-system/documentatie/assets/blog_toegankelijker-foutmeldingen-date-picker.png)
+
+For a deep dive into asking for dates in forms, see the GOV.UK Design System pattern library: [Ask users for Dates](https://design-system.service.gov.uk/patterns/dates/).
+
+Don't force the user to enter information twice (like an email address) and remember in a multistep form entries given in previous steps.
+
+{: .note }
+This success criterion is added in [WCAG 2.2: 3.3.7 Redundant Entry](https://www.w3.org/WAI/WCAG22/Understanding/redundant-entry.html).
+
+## What information can I provide in advance to help the user?
+
+We’ve all been there: you enter a new password, click submit, and *then* you’re told what the password requirements are. You yell at the screen: “Tell me that beforehand!”.
+
+The blog post [*Blind people don't visit my website*](https://www.a11y-collective.com/blog/blind-people-dont-visit-my-website/) lists frustrations from web users, such as:
+*"Error messages that come afterwards and eventually don't explain what you need to change."*
+
+Don’t make users guess — offer clear help in plain language, at the right moment. Tell users in advance which documents they'll need, and which fields are or aren't required. Add descriptions to fields where necessary.
+
+## How to show which fields are required?
+
+If you only ask what's really needed, all fields could technically be required — but that’s not always the case.
+
+### What’s the best way to indicate required?
+
+Many websites use indicators like “required”, “optional”, or an asterisk (\*). But which is clearest? 
+
+Do user research for your language. For example in Dutch the word "optioneel" (optional) is hard to understand for people with low literacy, and "niet verplicht" (not required) is a better choice. This may be different for your language.
+
+Prefer **“required”** over an asterisk, because an asterisk assumes prior knowledge of the meaning of that icon. If you do use an asterisk, explain its meaning above the form.
+
+### Mark required or non-required fields?
```

Comment:
"Should I mark *required* or *non-required* fields?"

Other subheadings are sentences, so I think this one should be, too.

Reply:
Yes - the first two are sentences, and the 3rd is a sentence fragment. (An incompletely structured sentence.) I'm suggesting that they should all use complete sentences.